python-checks/src/main/java/org/sonar/python/checks/NonCallableCalledCheck.java

```diff
@@ -55,22 +54,20 @@ public void initialize(Context context) {
     });
   }
 
-  protected static String addTypeName(InferredType type) {
-    String typeName = InferredTypes.typeName(type);
-    if (typeName != null) {
-      return " has type " + typeName + " and it";
-    }
-    return "";
+  protected static String addTypeName(PythonType type) {
+    return type.displayName()
+      .map(d -> " has type " + d + " and it")
+      .orElse("");
   }
 
   public boolean isNonCallableType(PythonType type) {
     return type.hasMember("__call__") == TriBool.FALSE;
   }
 
-  public String message(InferredType calleeType, @Nullable String name) {
+  public String message(PythonType typeV2, @Nullable String name) {
     if (name != null) {
-      return String.format("Fix this call; \"%s\"%s is not callable.", name, addTypeName(calleeType));
+      return String.format("Fix this call; \"%s\"%s is not callable.", name, addTypeName(typeV2));
```

Comment:
In currently used Java version `String.format` could be simplified by the `String.formatted` method:
```java
"Fix this call; \"%s\"%s is not callable.".formatted(name, addTypeName(typeV2))
```